### PR TITLE
[KDESKTOP-50] Fix cancel hydration menu

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

File: src/libsyncengine/syncpal/syncpal.h [hunk 1]
```diff
@@ -202,9 +202,7 @@ class SYNCENGINE_EXPORT SyncPal : public std::enable_shared_from_this<SyncPal> {
         ExitCode cancelDlDirectJobs(const std::list<SyncPath> &fileList);
         ExitCode cancelAllDlDirectJobs(bool quit);
         ExitCode cleanOldUploadSessionTokens();
-        inline bool isDlOngoing(const SyncPath &localPath) {
-            return _syncPathToDownloadJobMap.find(localPath) != _syncPathToDownloadJobMap.end();
-        }
+        bool isDlOngoing(const SyncPath &localPath);
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/syncpal/syncpal.h:205]: Would it be possible to find a more suggestive name? Does `Dl` stand for `Download`?

File: src/libsyncengine/syncpal/syncpal.cpp [hunk 2]
```diff
@@ -1349,6 +1349,20 @@ ExitCode SyncPal::cleanOldUploadSessionTokens() {
     return ExitCodeOk;
 }
 
+bool SyncPal::isDlOngoing(const SyncPath &localPath) {
+    if (_syncPathToDownloadJobMap.find(localPath) != _syncPathToDownloadJobMap.end()) {
+        return true;
+    }
+
+    for (const auto &syncPathToDownloadJobMapIt : _syncPathToDownloadJobMap) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/syncpal/syncpal.cpp:1357]:
```suggestion
    for (const auto &[path, _]: _syncPathToDownloadJobMap) {
        if (CommonUtility::isSubDir(localPath, path)) {
            return true;
        }
    }
```

File: src/libsyncengine/syncpal/syncpal.cpp [hunk 3]
```diff
@@ -1349,6 +1349,20 @@ ExitCode SyncPal::cleanOldUploadSessionTokens() {
...
+        if (CommonUtility::isSubDir(localPath, syncPathToDownloadJobMapIt.first)) {
+            return true;
+        }
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/syncpal/syncpal.cpp:1360]:
```suggestion
    for (const auto &[path, _]: _syncPathToDownloadJobMap) {
        if (CommonUtility::isSubDir(localPath, path)) {
            return true;
        }
```

**Review by luc-guyot-infomaniak — APPROVED** [review]
Only minor comments. Approved.

**Review by ClementKunz — APPROVED** [review]

File: src/libsyncengine/syncpal/syncpal.h (hunk 1 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/syncpal/syncpal.h]: Yes. Renamed.

File: src/libsyncengine/syncpal/syncpal.cpp (hunk 3 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/syncpal/syncpal.cpp]: Done

**Pull request merged**